docs/monitoring.md

```diff
@@ -0,0 +1,143 @@
+### CLI
+The unique name for the dahsboard will be printed on the console. This dashboards can be accessed through cloud monitoring custom dashboards page.
+
+## Metrics
+
+### Per Shard or Non-sharded Migration Dashboard
+The following is a list of important metrics to track during a migration:
+- Dataflow Workers CPU Utilisation
+- Datastream Throughput
+- Datastream Unsupported Events
+- Pubsub Age of oldest unacknowledged message
+- Spanner CPU Utilisation
+- Spanner Storage
+
+Below is an exhaustive list of all metrics in the dashboard and their usage.
+#### Dataflow
+
+| Metric | Description | Importance |
+|-----------|-----------|-----------|
+| Worker CPU Utilization | Shows the CPU Utilization of the 50th, 90th and 100th percentile worker| Used to identify if the pipelines is over or under scaled based on the value of CPU Utilization| 
+| Worker Memory Utilization | Shows the Memory Utilization of the 50th, 90th and 100th percentile worker| Used to identify if the health of the pipeline based on the value of Memory Utilization| 
+| Worker Max Backlog Seconds | Shows max time required to consume the largest backlog across all stages for each dataflow worker | Used to identify if the pipelines is over or under scaled| 
+
+
+#### Datastream
+
+| Metric    | Description | Importance |
+|-----------|-----------|-----------|
+| Throughput(events/sec) | Shows the average events processed/sec generated at source | Used to track if data is being transferred from source to GCS Bucket| 
+| Unsupported Events | Source events unsupported by Datastream | Used to identify if there is any data that can't be transferred by datastream due to a correctness issue| 
+| Total Latency | 50th and 90th percentile of time taken from event being written at source to being written by Datastream to GCS | Indicator of datastream being overloaded | 
+
+
+#### GCS Bucket
+
+| Metric | Description | Importance |
+|-----------|-----------|-----------|
+| Total Bytes | Shows the total bytes written to the GCS Bucket| Used to check if data is succesfully being written to the GCS Bucket| 
+
+
+#### Pubsub
+
+| Metric | Description | Importance |
+|-----------|-----------|-----------|
+| Published message count | Number of messages published by the GCS bucket to the Pub | Indicates total files in staging which need to be processed | 
+| Age of Oldest Unacknowledged Message | Age of the oldest unacknowledged message in the subscription | Used to determine if starvation of dataflow resources is taking place |
+
+
+#### Spanner
+
+| Metric | Description | Importance |
+|-----------|-----------|-----------|
+| CPU Utilisation | CPU Utilisation of spanner database and instance| Used to track if spanner is overloaded and requires more or less nodes | 
+| Storage | Storage of spanner database and instance | Used to track how the data is growing as the migration proceeds |
+
+
+### Aggregated Dashboard
```

Comment:
nit - move the aggregated dashboard above as it might be what more people will look at.

Reply:
Aggregated dashboard is only relevant for a sharded migration - I think it should be below the dashboard that is generated for every migration